### ISSUE TITLE
Cross-Project-Communication and community: Creation of Mailinglists?

### ISSUE THREAD

**Issue by jdittrich** [opensourcedesign/organization#69]
**Use Case**
* As a UX/Design/Usability… person from some project I would like to communicate with other open source designers
* As Opensourcedesign, we want to support collaboration and communication across project boundaries.

**Problem:** 
* Currently, open source designers will probably connect mainly in their own projects.
* Opensourcedesign currently offers and supports no such possibilities. When we talk here, we are mainly concerned with *our* organization, less with actual design problems from other projects. 

**Possible Solution:**
Can we create a/some mailinglists for that purpose? 

**Possible Challanges:**
* We would add a channel to our infrastructure
* Most projects can easily create an additional list, but it would be seen as theirs. The list should be on neutral space BUT I don't know if running our own mailman is cool.

**Comment by bnvk** [comment]
@jdittrich maybe I'm not understanding what you're describing. In *this* repo 
[organization](https://github.com/opensourcedesign/organization) it is a
conversation by OSD about itself- so maybe that's your comment

> When we talk here, we are mainly concerned with *our* organization, less with 
> actual design problems from other projects. 

The [jobs repo](https://github.com/opensourcedesign/jobs) has lots of
conversations between designers and other projects. Are you unaware this
going on or does it not resemble the problem you are talking about? If it's
the former, you need to "subscribe" to the repo to get notified (by email)
of conversations.

> Can we create a/some mailinglists for that purpose? 

I don't think traditional mailing lists make sense for OSD and the fact our 
repos are already providing mailing lists of sorts and

> * We would add a channel to our infrastructure

Yah, another *email* based channel, but it would lower the barrier of
needing to create a Github account.

I just setup a Lounge instance ( https://thelounge.github.io - decently
friendly web IRC client) so people can login / or leave their account idling. 
The Lounge is a fork of **Shout** which I had been working on the [Slick](https://github.com/opensourcedesign/slick) theme, 
which I was aiming to bring closer to Slack and should resume

**Comment by jdittrich** [comment]
>  [Jobs] Are you unaware this going on or does it not resemble the problem you are talking about?

I am aware of the repo, but only a few things of what I exchange about with other UX designers or researchers are "Jobs". Recently, I asked or answered (on other lists) how people do remote interviews, how one can analyze interviews in LibreOffice. I may want to ask if there is a good open video editor to edit screen records from my research or how a designer can involve developers. Those 

I can open github issues on that, but…
a) People who join need a developer-y github-account and can't say: "cool, I try that". 
b) I have doubt that issues are good for things that don't have a definition of done. (Some are fine, but many are a problem)

**Comment by bnvk** [comment]
> I may want to ask if there is a good open video editor to edit screen
> records from my research or how a designer can involve developers. 

Ok. I did misunderstood.
 
> a) People who join need a developer-y github-account and can't say: "cool, I try that". 

True. Mailmain (and mailing lists in general) are also an exclusive user 
demographic.

> b) I have doubt that issues are good for things that don't have a
> definition of done. 

This was an intentional choice from when I started creating all the repos. 
Your question above is a great question, but isn't it better to collect things 
like this on [resources](http://opensourcedesign.net/resources/) than have them 
buried (or simply only communicated by a few people) in a mail archive?

Of course there are questions that won't fit into that page (or elsewhere), 
but as far as "general discussion" with all the open source projects- do we
want to encourage that?

**Comment by simonv3** [comment]
If the chat room gets a bit more juice, wouldn't that be a good venue for asking these kinds of questions? I've found that with matrix scroll back and such is a much smaller problem.

**Comment by jdittrich** [comment]
> Mailmain (and mailing lists in general) are also an exclusive user demographic.

True, too. But they are a standard tool for communication and used by non-programmers too (at least in the form of google/yahoo groups). Thus, exclusive, yes (what is not) but known to work for many who are not us. 

> Your question above is a great question, but isn't it better to collect things like this on [resources]…

Unlikely. I (and most other people I know) write stuff like "Adding to Alice’s post above, I’d like to suggest to try LibreOffice Writer to do Y. Read paper X, which walks you through the basics.  " on Mailinglists – that would not be a resource one would add to the page. 

 > If the chat room gets a bit more juice…

I dislike them for such discussions (single threaded!), but  they seem to work for some. 

I ’m not saying that a mailinglist is the ideal way but I am concerned that the non-standard solutions we pursue can be pretty tricky for newcomers and/or peripheral participants (e.g. interested in OSD, but not taking part in org discussions like logo choice, venue selection etc.).

**Comment by bnvk** [comment]
> "Adding to Alice’s post above, I’d like to suggest to try LibreOffice Writer
> to do Y. Read paper X, which walks you through the basics. 

You're right that's not a "resource" but you might be missing my point- why 
not create a "Tips" page (repo) on the website as it's still helpful info :)

> I dislike them for such discussions (single threaded!), but they seem to
> work for some.

Good point. Single threaded is a short coming. Both chat & email also have the 
shortcoming of high signal to noise for hearing the question / finding the 
answer later like "Try LibreOffice Writer to do..." 

If what you want is an official OSD channel for "general discussion" which is 
inclusive to people not comfortable with creating a Github account, but are 
curious about OSD and prefer email, OK.

Personally, I don't think OSD should host such a forum. I'm having a hard 
enough time keeping with up with things and some barriers to entry are useful.

**Comment by jancborchardt** [comment]
We specifically decided against mailing lists in the beginning because they have several problems:
- they usually have bad interfaces where old messages are not easily findable
- they are not well integrated with other things like Github issues
- conversations on there always end up in banter and nothing gets done

This is why we chose Github issues. This platform also gets people used to the space where developers and open source projects usually work.

Sure, we could additionally set up something like [Discourse](http://www.discourse.org/) – but that would essentially be used for the _same_ thing that we use Github issues for, and just spread our channels thin.

**Comment by elioqoshi** [comment]
Whatever we choose, GitHub Issues, Mailing Lists, Discourse or anything else, will always have a barrier of involvement for a specific demographic, as these all cater to slightly different audiences. And since we have to reside with one solution, we can probably never fix this problem to its full extent.

One solution might be offering help@opensourcedesign.net for people who might be scared of GitHub but want to ask questions anyway. That is if we want to attract designers with no open source background.

We can also do whatcanidoforosd.net which basically gives you contribution suggestions on how one can contribute at OSD. Check out:

whatcanidoformozilla.org
whatcanidoforfedora.org

**Comment by jdittrich** [comment]
> We specifically decided against mailing lists …
> they usually have bad interfaces where old messages are not easily findable
> they are not well integrated with other things like Github issues

agree, that are good points.

> conversations on there always end up in banter and nothing gets done

Sometimes directly doing something is not the point – if I ask for advice on a problem or different views, there is not necessary something to be done, finished or to be finally resolved. This does not mean it is »banter«.

> And since we have to reside with one solution, we can probably never fix this problem to its full extent.

Good point.

Recently, I was talking with the people who ran openusability (@guiguru, @bbalazs ) and reading some interesting discussions on mailinglists on design research/usability recently. I am unsure what can be done to improve opensourcedesign for people gearing more to the conceptual/research-y spectrum of design. 

The mailinglist was merely a proposed way (and I share the views of the problems they entail and I can gladly close the suggestion); the actual challenge for me would be of how to exchange across projects and about problems that are not necessarily [ "resolvable"](https://en.wikipedia.org/wiki/Wicked_problem).

**Comment by jancborchardt** [comment]
@elioqoshi that’s why we need a better »Get involved« page – see https://github.com/opensourcedesign/organization/issues/54

**Comment by studiospring** [comment]
I have tried to outline the nature of OSD's communication needs in the following diagram. 
![osd_communication](https://cloud.githubusercontent.com/assets/4901306/23883873/614c2752-08b5-11e7-9b6d-07fc7b447978.png)

It only takes a short leap of logic from here to see that GitHub issues really does not suit the kinds of discussion that designers have.

A few things to note:
- it is simplistic (it only considers designers and devs as stakeholders). 
- you could be a unicorn.
- I really should have broken up project specific and agnostic characteristics in to 2 different diagrams.
- I have painted in broad strokes and generalisations.
- I can make the original svg files available on request (GH does not support svg).
- project specific discussion tools really should be outside the scope of this discussion since they need to be an integral part of a software management tool like GitHub or Phabricator, what I call a Collaborative Design and Development Environment (CDDE).

Here is the current state of play.
![communication_matrix](https://cloud.githubusercontent.com/assets/4901306/23883942/a06eda7e-08b5-11e7-8415-7531ffc46d75.png)

The next steps are to:
1. decide what kinds of discussions OSD actually wants to support.
2. determine the requirements for those types of discussion.
3. find the tools that best meet the above requirements.

**Comment by guiguru** [comment]
hey everybody, peter sikking from openUsability here.

what @studiospring contributes above is really impressive _(I do not get to say that very often)_.
get the collaboration with developers off the playing fields they own.
a couple of remarks:

I am pretty sure @jdittrich’s original topic is about a designer community where we get to support each other on a moral level, as in: “I saw your outcry of project frustration _[there needs to be a place for that]_. don’t let the developers bully you. now is the moment you set an ultimatum.”

a designer who is not _on_ a project is, ehm, not a designer on the project. this severely limits the problem-solving capabilities of this designer—to the point of being just another innocent bystander. one exception I see to this is a mentoring relationship, whether long-term or ad hoc, senior to junior.

design is: 1) a structured process; 2) a specification; 3) shipped results.

any component missing: it is not design. 2) is crucial for getting to 3). thus designer to developer work is not built on ideas—ideas are a dime a dozen—but on a specified solution, that is refined as feedback comes in. being a designer is being a translator between the conflicting parties (product, users, tech); the spec is where translation for tech (developers ) is done. I would really like to see this to have an important role in the orange box (Collaborative Design and Development Environment) above.

**Comment by jdittrich** [comment]
@guiguru:

> designer community where we get to support each other on a moral level…

Also that. I was thinking along the lines of a ["community of practice"](https://en.wikipedia.org/wiki/Community_of_practice) (A group of people sharing information and experiences to develop personally and professionally).  Of which "how do I deal with that conflict?" is as much part as "I tried paper prototyping recently, here are my pros and cons so far!" or "I want to provide a coded mock, can so help with CSS?"

Such a community of practice I would see in groups like [Anthrodesign](https://groups.yahoo.com/neo/groups/anthrodesign/info) and also here in OSD – my intention was not saying we don't have communication in the sense of a community of practice but to strengthen it.

@studiospring Thanks for the visualizations!

**Comment by studiospring** [comment]
Kind words indeed, @guiguru. I suspect a Collaborative Design & Development Environment will remain a dream until we understand the systemic issues (described briefly [here](https://www.quora.com/Why-does-open-source-software-usually-have-poor-UI-and-UX-design-What-can-be-done-to-improve-this) as the value exchange system) preventing its creation/adoption. Even then, open source designers alone may not have the resources to tip the value exchange system in the users' favour.

The first step? More research! Watch this space!

To get back on topic (!) though, I would like to see this kind of discussion done in a discussion forum and leave GH Issues for the code issues that it was intended for.

**Comment by simonv3** [comment]
Does an online discussion forum (like Discourse) or a listserv fix the problems that GitHub has? Is there there a real difference between a discussion forum and the issues on a GitHub repository except for naming? Naming could be a significant reason as so much of our interaction is based framing, but I don't know if switching from GitHub to an other (currently suggested) web based forum will alleviate or fix our problems. 

@studiospring that research and diagramming is excellent. Thank you so much for doing that. I think both you and @jdittrich are getting at a core issue with OSD - that we're catering to developers who want more design, rather than designers who want to work on open source. I don't know what the fix for that is.

**Comment by studiospring** [comment]
@simonv3 I think the bottom diagram makes clear that there are platforms for (open source) developers to discuss and collaborate, but there are none for designers or for designer-dev collaboration. We are using GH Issues for something it wasn't intended for. Personally, I think the differences between a discussion forum and GH Issues is more than semantics. Here are just a few problems with Issues that could be fixed immediately with almost any decent forum:

- not mobile friendly.
- SEO???
- GH account required. This is partly an image problem. "What is GitHub?", "GitHub is for developers!"
- general UX: findability, organisation, notification, poor svg support, threading...
- not open source.

I do not think moving to a forum will alienate developers, especially if it is integrated with GitHub (Discourse, hint, hint).

A designer-dev collaboration tool is, of course, a whole other beast.

**Comment by simonv3** [comment]
Take these responses with a grain of salt - I don't think GitHub is perfect for what we're doing.

1. I regularly use GitHub on my phone so I don't feel like it's not mobile friendly. Could you give me an example of where it's lacking?
2. Most of my searches for specific libraries and such when I'm dealing with bugs and such land me on exactly the GitHub issue I need. Are you afraid that "OSD" isn't findable, or that GitHub issues specifically aren't findable?
3. You'll need an account for any forum software you set up. Granted, you can brand that different from how GitHub brands it. But at the same time GitHub is where the open source _software_ gets made. If you're going to be participating in making that software it's a platform you'll need to know (or one of its mimics). I don't see asking that a designer have a GitHub account is a bad thing - they'd probably need one in the end anyway.
4. Yeah. Findability across multiple repositories is GitHub's main flaw for me at the moment. Threading is fair critique.
5. Fair.

Here's the thing: to me a huge part of Open Source Design is that people try the things that they think they want to try. If someone sets up a Discourse forum for us to use, and it takes off, then wahay, that's great. I'll create an account and participate right away. Don't rely on others to do the work for you though because people only have so much time they can contribute.

Would creating a Discourse forum also solve #48 - handling small UX jobs? Would it allow for better communication for local meetups #58? Would it be possible to have a sub-forum for all the OSD conversation so that it is easy to find and in one place #62 #64 #65 #66? What about niche conversation groups (like UX, research, etc). 

I think the question is probably yes to all of the above.

**Comment by studiospring** [comment]
@simonv3 
1. This is what GH looks like for me (quick test in browser).
![responsive](https://cloud.githubusercontent.com/assets/4901306/23930036/9134ce90-0975-11e7-8f4c-eb9ed9019985.jpg)
2. SEO is important for designers who do not know about open source or GH to discover serendipitously or actively through a search engine.
3. This may not be a huge problem, but I think a regular forum is a lower barrier to entry for non-OS, non-developer designers than asking them to sign in to a developers' "development platform". Here is the GH home/sign up page. Putting myself in their shoes: "I'm a designer! Why would I want to sign up to a developer's platform? I want to talk with designers!"
![gh](https://cloud.githubusercontent.com/assets/4901306/23930265/df65fbb0-0976-11e7-9f35-9774405a7fb9.jpg)

 I also think decoupling OSD from GH is also a good thing.

Regarding your other comments, I am happy to put in some work (for now). Consider this the research phase (and we need 2000 stars for free Discourse hosting)!

**Comment by bnvk** [comment]
Perhaps setting up a general "Discussion" repo will help (for now) to meet
@jdittrich original question? Otherwise, i'm all for experimentation- if enough 
of you want a Discourse instance for OSD, someone should go set one up :)

> 1. This is what GH looks like for me (quick test in browser).

@studiospring that's not the mobile version of Github. Maybe, something's wonky
with your mobile device / browser?

> 2. SEO is important for designers who do not know about open source or GH 

I'd rather see our *website* become rich in content, have good design and SEO- 
not expect GitHub issues to come up in search engines. Making a great dynamic 
website has always been a big part of what I've wanted OSD to become. 

> ...a regular forum is a lower barrier to entry for non-OS, non-developer 
> designers than asking them to sign in to a developers' "development platform".

There's a strong argument that designers congregate on visual platforms and
don't feel comfortable on text based forums. As per the diagram of 
"Project Agnostic" designer platforms here's some:

- Dribble - https://dribbble.com
- Behance - https://www.behance.net
- Forrst - http://zurb.com/forrst (kinda defunct)

> I also think decoupling OSD from GH is also a good thing.

Yes! The *how* is crucial. IMHO this should be done carefully and not
require entire "island-like" platforms like Discourse. I've started a
separate issue as this is bigger discussion than just "a forum for discussion"
with my proposed path towards this goal: 

https://github.com/opensourcedesign/organization/issues/73

> Regarding your other comments, I am happy to put in some work (for now). 
> Consider this the research phase

@studiospring great :D Are you developer- back-end, front-end? Or more of a 
designer?

**Comment by studiospring** [comment]
@bnvk You're right, for some reason, GH is not responsive on my Firefox, but is on Chromium. Apologies for the confusion.

> IMHO this should be done carefully and not require entire "island-like" platforms like Discourse. 

I'm not sure what you mean by this. It seems Discourse, or any other forum can be [integrated in to an existing site](https://meta.discourse.org/t/deeply-integrating-discourse-into-website/44214).

I started in front end, then back end, now UX designer.

**Comment by grahamperrin** [comment]
> … [just setup a Discourse site.](https://github.com/opensourcedesign/organization/issues/73#issue-214443776)

+1

> … I'm not saying general discussions are bad- they are just not my interest. 
> 
> Thus if enough others *do* want to do that, setup a Discourse forum or something similar :D

> … [something like Discourse – but that would essentially be used for the same thing that we use Github issues for,](https://github.com/opensourcedesign/organization/issues/69#issuecomment-285795767) … 

With respect: the essences can be quite different. 

> … [Does an online discussion forum (like Discourse) … fix the problems that GitHub has?](https://github.com/opensourcedesign/organization/issues/69#issuecomment-286582566) … 

In my experience: Discourse can fix, negate or – most often, **prevent** ninety-nine percent of such problems. 

> … [I do not think moving to a forum will alienate developers, especially if it is integrated with GitHub (Discourse, hint, hint).](https://github.com/opensourcedesign/organization/issues/69#issuecomment-286600297) …

+1

> … [If someone sets up a Discourse forum for us to use, and it takes off, then wahay, that's great.](https://github.com/opensourcedesign/organization/issues/69#issuecomment-286609405) …
> 
> Would creating a Discourse forum also solve …
> 
> … probably yes to all of the above.

Almost certainly _yes_. 

There's great [praise for Discourse](https://meta.discourse.org/c/praise). [From me](https://meta.discourse.org/t/vote-for-discourse-in-the-cms-critic-awards-2016/50898/4?u=grahamperrin): 

> … it seems that on Twitter, I describe something as excellent just once a year …

– that one thing in 2016 was Discourse (but with the peculiarities of Twitter, the relevant tweet is no longer found). 

By near coincidence – I was unaware of _Open Source Design_ when I began this topic: 

- [Discourse for discussions: superiority to e.g. GitHub - praise - Discourse Meta](https://meta.discourse.org/t/discourse-for-discussions-superiority-to-e-g-github/58937?u=grahamperrin)

> … Last but not least (this topic should not be an exercise in bashing GitHub): …

For countless reasons, my strongest recommendation is to use Discourse for discussions. 

> … [need 2000 stars for free Discourse hosting](https://github.com/opensourcedesign/organization/issues/69#issuecomment-286619516) …

Good point. Is it necessary to manually calculate the sum of the separate counts of stars at https://github.com/opensourcedesign or am I looking in the wrong place?

**Comment by m1guelpf** [comment]
Just seen this http://uid8app.com

**Comment by jdittrich** [comment]
@m1guelpf: Thanks for posting. Is this an opensource tool and/or a communication platform?

**Comment by m1guelpf** [comment]
@jdittrich I don't think so, sorry.

**Comment by grahamperrin** [comment]
Given the twelve subscriptions that are automated when https://github.com/opensourcedesign is joined –

1. https://github.com/opensourcedesign/resources/subscription
2. https://github.com/opensourcedesign/patterns/subscription
3. https://github.com/opensourcedesign/jobs/subscription
4. https://github.com/opensourcedesign/fonts/subscription
5. https://github.com/opensourcedesign/events/subscription
6. https://github.com/opensourcedesign/slick/subscription
7. https://github.com/opensourcedesign/opensourcedesign.github.io/subscription
8. https://github.com/opensourcedesign/organization/subscription
9. https://github.com/opensourcedesign/funding/subscription
10. https://github.com/opensourcedesign/icons/subscription
11. https://github.com/opensourcedesign/foss-contributors-checklist/subscription
12. https://github.com/opensourcedesign/talks/subscription

– and looking ahead to a discussion forum … for starters I imagine that _organisation_ (or _organization_) can be within a _lounge_ area. 

Thoughts? Does the **lounge** metaphor work well enough? 

From https://github.com/opensourcedesign/organization/issues/69#issuecomment-286310583 above: 

> 1. decide what kinds of discussions OSD actually wants to support.

Re: the four links at the head of http://opensourcedesign.net/ maybe have an area for discussion of **articles**. 

(Some circularity: http://opensourcedesign.net/2015/11/21/5-steps-to-design-a-ux-that-people-love (originally [5 steps to design a UX that people love – Open Source Design – Medium](https://medium.com/open-source-design/open-source-design-just-steal-everything-5c44ca793721)) is partly about Loomio, a tool for decision-making.)

**Comment by grahamperrin** [comment]
> 1. decide what kinds of discussions OSD actually wants to support.

From the **Welcome to Open Source Design** at https://github.com/opensourcedesign/opensourcedesign.github.io/blob/master/CONTRIBUTING.md, these six points can help with a round of decision-making: 

>> … determine if you want to contribute to the OSD project itself and/or our community, or if you want to contribute design related resources, job offers, events.
>> 
>> - General Design
>> - Web Design
>> - Documentation
>> - Web Development
>> - Fundraising
>> - Event Organizing

**Comment by simonv3** [comment]
This overview of how many activists groups are organizing seemed really relevant to me: http://civichall.org/civicist/movements-organize-now-notes-challenges-rapid-growth/

**Comment by elioqoshi** [comment]
These Meta Discussions are really great but I think it's getting quite a bit out of hands to follow :D
Maybe we should have these discussions at the summit or at least the next video meeting? I think we might miss out on a lot of opinions, as all these great topics might be hard to digest for members. 

What do you think?

**Comment by jdittrich** [comment]
I’d love to discuss those at a summit or meeting; however, I would not discourage having the discussions online, too (Dont know if that is what you would suggest).

**Comment by simonv3** [comment]
Yeah, I think we can do both? I haven't seen as much life in OSD as we've had since we started these meta discussions :)

**Comment by grahamperrin** [comment]
> … next video meeting …

Will the date be added to https://github.com/opensourcedesign/events/issues/49#issue-185188141 or is there another calendar? Something I saw this week (I can no longer find it) made me wonder whether there's already a CalDAV service for something related to OSD.

**Comment by grahamperrin** [comment]
Big thanks @simonv3 for adding the link to #49. 

I did seek _nextcloud_ before asking the question but `<deep sigh>`GitHub`</deep sigh>` fails to find what's required. https://github.com/search?q=org%3Aopensourcedesign+nextcloud for example. 

> … out of hands to follow :D …

Yeah! I'd like to add some things that are relevant to this issue but right now I'm suffering from GitHub horror `:-/`
